autometa/config/databases.py

```diff
@@ -66,230 +97,504 @@ class Databases:
-                        raise ConnectionError(f'{option} download failed')
-                    ftp.quit()
+
             logger.debug(f'UPDATE: (ncbi,{option}): {outfpath}')
             self.config.set('ncbi', option, outfpath)
-        # Extract/format respective NCBI files
-        self.extract_taxdump()
-        self.format_nr()
 
-    def update_markers(self, options):
-        """Update single-copy markers hmms and cutoffs.
+            if self.dryrun:
+                return
+
+            with FTP(host) as ftp, open(outfpath, 'wb') as fp:
+                ftp.login()
+                logger.debug(f'starting {option} download')
+                result = ftp.retrbinary(f'RETR {ftp_fpath}', fp.write)
+                if not result.startswith('226 Transfer complete'):
+                    raise ConnectionError(f'{option} download failed')
+                ftp.quit()
+        if 'taxdump' in options:
+            self.extract_taxdump()
+        if 'nr' in options:
+            self.format_nr()
+
+    def press_hmms(self):
+        """hmmpress markers database files.
+
+        Returns
+        -------
+        NoneType
+
+        """
+        hmms = (os.path.join(self.markers_dir, fn) for fn in os.listdir(self.markers_dir) if fn.endswith('.hmm'))
+        hmm_search_str = os.path.join('autometa/databases/markers/','*.h3*')
+        pressed_hmms = glob(hmm_search_str)
+        pressed_hmms = set(os.path.realpath(os.path.splitext(fp)[0]) for fp in pressed_hmms)
+        for fp in hmms:
+            if fp in pressed_hmms:
+                continue
+            hmmer.hmmpress(fp)
+
+    def download_markers(self, options):
+        """Download markers database files and amend user config to reflect this.
 
         Parameters
         ----------
-        options : set
-            Description of parameter `options`.
+        options : iterable
+            iterable containing options in 'markers' section to download.
 
         Returns
         -------
         NoneType
+            Will update provided `options` in `self.config`.
 
         Raises
         -------
         ConnectionError
             marker file download failed.
 
         """
-        if not os.path.exists(self.markers_dir):
-            os.makedirs(self.markers_dir)
         for option in options:
-            url = DEFAULT_CONFIG.get('database_urls', option)
+            # First retrieve the markers file url from `option` in `markers` section
+            url = self.config.get('database_urls', option)
             if self.config.has_option('markers', option):
                 outfpath = self.config.get('markers', option)
             else:
                 outfname = os.path.basename(url)
                 outfpath = os.path.join(self.markers_dir, outfname)
+
             if self.dryrun:
                 logger.debug(f'UPDATE: (markers,{option}): {outfpath}')
                 self.config.set('markers', option, outfpath)
                 continue
```

Comment:
Instead of `continue`, shouldn't this be `return`? Otherwise if `self.dryrun == True`, then we still go ahead and download the markers.

Reply:
This is looping through the `options`, so continuing will skip the downloading and amend the `config` with the option.